src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java

```diff
@@ -0,0 +1,68 @@
+package school.hei.patrimoine.compiler;
+
+import static javax.tools.ToolProvider.getSystemJavaCompiler;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.function.Function;
+import lombok.SneakyThrows;
+
+public class CasFileCompiler implements Function<String, Class<?>> {
+
+  private static final String COMPILE_DIR_NAME =
+      System.getProperty("user.home") + "/.patrimoine/compile";
+  public static final String DEPENDENCY_JAR_PATH =
+      System.getProperty("user.home") + "/Downloads/drive";
+
+  static {
+    new File(COMPILE_DIR_NAME).mkdirs();
+  }
+
+  private static String getClassNameFromPath(String filePath) {
+    return Path.of(filePath).getFileName().toString().replace(".java", "");
+  }
+
+  @SneakyThrows
+  @Override
+  public Class<?> apply(String filePath) {
+    var ioDirPath = Path.of(COMPILE_DIR_NAME);
+    var sourcePath = Path.of(filePath);
+    var className = getClassNameFromPath(filePath);
+
+    compile(ioDirPath, sourcePath);
+    return loadClass(className);
+  }
+
+  private Class<?> loadClass(String className)
+      throws ClassNotFoundException, MalformedURLException {
+    var ioDirPath = Path.of(COMPILE_DIR_NAME);
+
+    return Class.forName(
+        className, true, URLClassLoader.newInstance(new URL[] {ioDirPath.toUri().toURL()}));
+  }
+
+  private void compile(Path ioDirPath, Path sourcePath) {
+    var compiler = getSystemJavaCompiler();
+    var PATRIMOINE_JAR_PATH = Path.of(DEPENDENCY_JAR_PATH).resolve("patrimoine-1.0-SNAPSHOT.jar");
```

Comment:
magic variable patrimoine-1.0 smthg, statically name it

Reply:
done